indykite/resource_entity_matching_pipeline.go

```diff
@@ -102,14 +127,28 @@ func resourceEntityMatchingPipelineBuild(
 	_ *ClientContext,
 	builder *config.NodeRequest,
 ) {
-	sourceNoteTypes := rawArrayToTypedArray[string](data.Get(entityMatchingPipelineSourceNodeFilterKey))
-	targetNodeTypes := rawArrayToTypedArray[string](data.Get(entityMatchingPipelineTargetNodeFilterKey))
+	cfg := &configpb.EntityMatchingPipelineConfig{}
 
-	cfg := &configpb.EntityMatchingPipelineConfig{
-		NodeFilter: &configpb.EntityMatchingPipelineConfig_NodeFilter{
-			SourceNodeTypes: sourceNoteTypes,
-			TargetNodeTypes: targetNodeTypes,
-		},
+	// Check if SourceNodeFilterKey and TargetNodeFilterKey are available
+	sourceNodeTypes, hasSourceKey := data.GetOk(entityMatchingPipelineSourceNodeFilterKey)
+	targetNodeTypes, hasTargetKey := data.GetOk(entityMatchingPipelineTargetNodeFilterKey)
+
+	if hasSourceKey && hasTargetKey {
```

Comment:
This `if` is not necessary, it will be always true as they are required

Reply:
indeed